test/reduce.js

```diff
@@ -291,7 +327,7 @@ module.exports = function reduce_test(testcase, minify_options, reduce_options)
             }
 
             // replace this node
-            var newNode = new U.parse(REPLACEMENTS[node.start._permute % REPLACEMENTS.length | 0], {
+            var newNode = U.parse(REPLACEMENTS[node.start._permute % REPLACEMENTS.length | 0], {
```

Comment:
Fixed typo. I was surprised this had worked at all with `new`.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new
> The `new` keyword does the following things:
> 1. Creates a blank, plain JavaScript object;
> 2. Links (sets the constructor of) this object to another object;
> 3. Passes the newly created object from *Step 1* as the `this` context;
> 4. **Returns `this` if the function doesn't return its own object.**

But yeah, I didn't notice that either - sorry 😅

Reply:
Actually, that means we can optimise `new F(...)` ➡️ `F(...)` as long as:
- `typeof F() == "object"` always
- `!F.contains_this()`

Reply:
Worth a shot. There's always unanticipated found problems in practice. Does ufuzz generate such `new` cases with and without `this`?

Reply:
I believe it does:
https://github.com/mishoo/UglifyJS2/blob/b0040ba6540232687204d125eff021adf11c4efa/test/ufuzz/index.js#L691